src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -624,6 +624,40 @@ def b64_to_hex(s):
     return hex_data
 
 
+def todict(obj, post_processor=None):
+    """
+    Convert an object to a dictionary. Use 'post_processor(original_obj, dictionary)' to update the
+    dictionary in the process
+    """
+    from datetime import date, time, datetime, timedelta
+    from enum import Enum
+    if isinstance(obj, dict):
+        result = {k: todict(v, post_processor) for (k, v) in obj.items()}
+        return post_processor(obj, result) if post_processor else result
+    if isinstance(obj, list):
+        return [todict(a, post_processor) for a in obj]
+    if isinstance(obj, Enum):
+        return obj.value
+    if isinstance(obj, (date, time, datetime)):
+        return obj.isoformat()
+    if isinstance(obj, timedelta):
+        return str(obj)
+    # This is the only difference with knack.util.todict because for typespec generated SDKs
+    # The base model stores data in obj.__dict__['_data'] instead of in obj.__dict__
+    # We need to call obj.as_dict() to extract data for this kind of model
+    if hasattr(obj, 'as_dict') and not hasattr(obj, '_attribute_map'):
+        result = {to_camel_case(k): todict(v, post_processor) for k, v in obj.as_dict().items()}
+        return post_processor(obj, result) if post_processor else result
```

Comment:
A good question from @kairu-ms, what if the REST spec defines a `asDict` property?

Reply:
For now there is no such a property: https://github.com/search?q=repo%3AAzure%2Fazure-rest-api-specs%20asDict&type=code. If there is one in the future, we could rename it.